rust-escrow/stable-escrow/src/contract.rs

```diff
@@ -0,0 +1,123 @@
+use near_sdk::{
+    collections::UnorderedMap, env, json_types::U128, log, near_bindgen, serde_json::json,
+    AccountId, Balance, Promise,
+};
+
+use crate::consts::*;
+use crate::storage::*;
+
+impl Default for ConditionalEscrow {
+    fn default() -> Self {
+        env::panic_str("ConditionalEscrow should be initialized before usage")
+    }
+}
+
+#[near_bindgen]
+impl ConditionalEscrow {
+    #[init]
+    pub fn new(metadata: Metadata) -> Self {
+        if env::state_exists() {
+            env::panic_str("ERR_ALREADY_INITIALIZED");
+        }
+
+        Self {
+            deposits: UnorderedMap::new(b"r".to_vec()),
+            metadata: Metadata {
+                dao_name: None,
+                ..metadata
+            },
+        }
+    }
+
+    /**
+     * Called by anyone only once, this creates the NEP141 own token to be transfered in exchange of the stable NEP141 deposits
+     */
+    pub fn publish(&mut self, sender_id: AccountId, amount: Balance) {}
```

Comment:
@susguzman I think we'll need to call this function after deployment in order to create the own NEP141 that'll be transferred in exchange of the NEP141 stable coin.

Reply:
The Escrow now is a NEP141. So, the `publish` fn is not needed.